sdk/python/kfp/cli/compile_.py

```diff
@@ -133,14 +133,27 @@ def parse_parameters(parameters: Optional[str]) -> Dict:
     is_flag=True,
     default=False,
     help='Whether to disable type checking.')
+@click.option(
+    '--disable-caching',
+    is_flag=True,
+    default=False,
+    help="Disable task-level caching by default."
+)
 def compile_(
     py: str,
     output: str,
     function_name: Optional[str] = None,
     pipeline_parameters: Optional[str] = None,
     disable_type_check: bool = False,
+    disable_caching: bool = False,
 ) -> None:
     """Compiles a pipeline or component written in a .py file."""
+
+    if disable_caching:
+        os.environ['KFP_DISABLE_CACHING'] = 'true'
```

Comment:
same as above -- `KFP_EXECUTION_CACHING_DEFAULT`
with values of `enabled` and `disabled`

Reply:
Done, updated to have a CLI flag `--execution-caching-enabled-by-default` and an environment variable `KFP_EXECUTION_CACHING_ENABLED_BY_DEFAULT`, that take in values of `enabled` or `disabled`.